src/arithmetic/algebraic_expression.c

```diff
@@ -330,7 +330,8 @@ AlgebraicExpression **AlgebraicExpression_From_Query(const AST *ast, Vector *mat
         e = QueryGraph_GetEdgeByAlias(q, astEdge->ge.alias);
         assert(e);
 
-        GrB_Matrix mat = e->mat;
+        // TODO: we might want to delay matrix retrieval even further.
```

Comment:
Past here, would that make a difference?

Reply:
might be, in some cases...